Lib/test/test_peepholer.py

```diff
@@ -1071,7 +1071,57 @@ def test_no_unsafe_static_swap(self):
             ('POP_TOP', 0, 4),
             ('RETURN_VALUE', 5)
         ]
-        self.cfg_optimization_test(insts, insts, consts=list(range(3)), nlocals=1)
+        expected_insts = [
+            ('LOAD_CONST', 0, 1),
+            ('LOAD_CONST', 1, 2),
+            ('NOP', 0, 3),
+            ('STORE_FAST', 1, 4),
+            ('POP_TOP', 0, 4),
+            ('RETURN_VALUE', 5)
+        ]
+        self.cfg_optimization_test(insts, expected_insts, consts=list(range(3)), nlocals=1)
+
+    def test_dead_store_elimination_in_same_lineno(self):
+        insts = [
+            ('LOAD_CONST', 0, 1),
+            ('LOAD_CONST', 1, 2),
+            ('LOAD_CONST', 2, 3),
+            ('STORE_FAST', 1, 4),
+            ('STORE_FAST', 1, 4),
+            ('STORE_FAST', 1, 4),
+            ('RETURN_VALUE', 5)
+        ]
+        expected_insts = [
+            ('LOAD_CONST', 0, 1),
+            ('LOAD_CONST', 1, 2),
+            ('NOP', 0, 3),
+            ('POP_TOP', 0, 4),
```

Comment:
This is a cool benefit I didn't realize we'd also get! `LOAD_*/POP_TOP` pairs can be eliminated entirely.

Reply:
Ideally this would just become `LOAD_CONST NOP NOP STORE_FAST` though, right?

Reply:
Yes; I think that's an orthogonal improvement to the `LOAD/POP` elimination optimization, allowing it to ignore intervening `NOP`

Reply:
Maybe we can run the loop of optimization until there are no opcode changes.
(I didn't test test)